core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -885,19 +905,11 @@ public void setPinned(LockedInodePath inodePath, boolean pinned, long opTimeMs)
     } else {
       assert inode instanceof InodeDirectory;
       // inode is a directory. Set the pinned state for all children.
-      try (TempInodePathForDescendant tempInodePath = new TempInodePathForDescendant(inodePath)) {
+      try {
```

Comment:
Is this `try` extraneous?

Reply:
don't think so, the logic is that if a single `lockDescendantPath` failed, we should not try to  lock more descendants. Hence the `try` block outside of the loop.